File: src/Book.cpp
#include <iostream>
#include <string>
#include "./Author.cpp"

using namespace std;

class Book
{
private:
    string title;      // book name
    string ISBN;       // book number
    string genre;      // genre type
    bool availability; // bookavaiable
    // Author* author;    // auther defin

public:
    Book(string t, string i, string g, author nullptr)
    {
        title = t;
        ISBN = i;
        genre = g;
        // author = author;
        availability = true;
    };

    void borrowBook()
    {
        if (availability)
        {
            cout << "Book borrowed successfully by " << endl;
            availability = false;
        }
        else
        {
            cout << "Book is not available for borrowing." << endl;
        }
    }

    void returnBook()
    {
        availability = true;
        cout << "Book returned successfully." << endl;
    }

    void calculateFine()
    {
        cout << "Fine calculation ." << endl;
    }

    void setAuthor(Author* author) {
        this->author = author;
    }
};

File: src/Book.js
import Author from "./Author.js";

class Book {
    constructor(title, ISBN, genre, author = null) {
        this.title = title;
        this.ISBN = ISBN;
        this.genre = genre;
        this.author = author;
        this.availability = true;
    }

    setAuthor(author) {
        this.author = author;
    }
}

export default Book;
